ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -854,8 +892,14 @@ generateGhcLibCabal ghcFlavor = do
         , "homepage: https://github.com/digital-asset/ghc-lib"
         , "bug-reports: https://github.com/digital-asset/ghc-lib/issues"
         , "data-dir: " ++ dataDir
-        , "data-files:"] ++ indent dataFiles ++
-        [ "extra-source-files:"] ++ indent (ghcLibExtraFiles ghcFlavor \\ [hadrianGeneratedRoot ghcFlavor </> "ghcversion.h"]) ++
+        , "data-files:"] ++ indent (dataFiles ghcFlavor) ++
+        [ "extra-source-files:"] ++
+          indent (ghcLibExtraFiles ghcFlavor
+                    [ (hadrianGeneratedRoot ghcFlavor </> "ghcversion.h", Nothing)
+                    , (hadrianGeneratedRoot ghcFlavor </> "DerivedConstants.h", Just $ hadrianGeneratedRoot ghcFlavor </> "GhclibDerivedConstants.h")
```

Comment:
Was this just broken before? Afaict, we used `DerivedConstants.h` instead of `GhclibDerivedConstants.h`.

Reply:
we did use just `DerivedConstants.h` before. this means either it didn't matter if the build compiler version got substituted for it when building `ghc-lib` or, it didn't before get referenced during the building of `ghc-lib` as it does now. i think the latter is more probable.